File: src/mods/eos_mod.f90
!vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvC
!                                                                      C
!  Module name: EOS                                                    C
!  Purpose: Equation of state for gas and initial solids density       C
!                                                                      C
!  Author: M. Syamlal                                 Date: 29-JAN-92  C
!                                                                      C
!^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^C
module eos

  use amrex_fort_module, only : rt => amrex_real
  use iso_c_binding , only: c_int

contains

!vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvC
!                                                                      C
!  Function: EOSG                                                      C
!  Purpose: Equation of state for gas                                  C
!                                                                      C
!^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^C
  real(rt) function eosg (mw, pg, tg)

! Global Variables:
!---------------------------------------------------------------------//
    use constant, only: gas_const

    implicit none

! Dummy arguments
!---------------------------------------------------------------------//
    real(rt), intent(in) :: mw, pg, tg

    eosg = pg*mw/(gas_const*tg)
    return
  end function eosg

end module eos

File: src/mods/eos_mod.cpp
#include <eos_mod.H>

AMREX_GPU_HOST_DEVICE
amrex::Real idealgas (const amrex::Real& mw, const amrex::Real& pg, const amrex::Real& tg)
{
  return (pg*mw)/(8314.56*tg);
}



AMREX_GPU_HOST_DEVICE
amrex::Real sutherland (const amrex::Real& tg)
{
   // Dummy arguments
   //---------------------------------------------------------------------//

   // Gas viscosity   (in Poise or Pa.s)
   // Calculating gas viscosity using Sutherland's formula with
   // Sutherland's constant (C) given by Vogel's equation C = 1.47*Tb.
   // For air  C = 110 (Tb=74.82)
   //         mu = 1.71*10-4 poise at T = 273K

   const amrex::Real tg_scaled = tg/273.0;

   return 1.7e-5 * tg_scaled * std::sqrt(tg_scaled) * (383./(tg+110.));
}
